modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -6609,20 +6648,7 @@ protected String toMediaTypeSchemaName(String contentType) {
                         for (Entry<String, Header> headerEntry: encHeaders.entrySet()) {
                             String headerName = headerEntry.getKey();
                             Header header = ModelUtils.getReferencedHeader(this.openAPI, headerEntry.getValue());
```

Comment:
You've introduced a tiny bug by not calling this at the other place headerToCodegenParameter is called, therefore referenced headers do not generate correctly.
I'm providing a PR to fix this.